base_layer/core/src/base_node/sync/rpc/sync_utxos_task.rs

```diff
@@ -71,10 +75,15 @@ where B: BlockchainBackend + 'static
                     RpcStatus::general("DB failure when fetching header containing start index")
                 }
             })?;
+        let hash = msg
+            .end_header_hash
+            .clone()
+            .try_into()
+            .rpc_status_internal_error(LOG_TARGET)?;
```

Comment:
This isn't an internal error - this is valid user input
```suggestion
            .rpc_status_bad_request("Invalid length of end_header_hash")?;
```